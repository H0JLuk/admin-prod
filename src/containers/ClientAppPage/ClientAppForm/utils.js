import { notification } from 'antd';
import { addSettings, updateSettingsList } from '../../../api/services/settingsService';
import { designKeysForCheck, SETTINGS_TYPES } from './ClientAppFormConstants';

export const showNotify = (message, isError) => {
    const config = {
        duration: 5,
        placement: 'bottomRight',
    };

    if (isError) {
        notification.error({
            ...config,
            message,
        });
    } else {
        notification.success({
            ...config,
            message,
        });
    }

};

export async function createOrUpdateKey (changedParams) {
    const updateSettings = [];
    const addSettingsArr = [];

    changedParams.forEach(({ type, ...params }) => {
        type === SETTINGS_TYPES.EDIT
            ? updateSettings.push(params)
            : addSettingsArr.push(params);
    });

    updateSettings.length && (await updateSettingsList(updateSettings));
    addSettingsArr.length && (await addSettings(addSettingsArr));
}

export const checkExistDesignSettings = (settings) => {
    return designKeysForCheck.some(key => !Object.prototype.hasOwnProperty.call(settings, key));
};

export const getImage = (imageName) => {
    return `${process.env.PUBLIC_URL}/images/clientAppDesign/${imageName}.png`;
};


