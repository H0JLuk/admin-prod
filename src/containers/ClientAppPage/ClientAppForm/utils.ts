import { customNotifications } from '@utils/notifications';
import { ArgsProps } from 'antd/lib/notification';
import { addSettings, updateSettingsList } from '@apiServices/settingsService';
import { SETTINGS_TYPES } from './ClientAppFormConstants';
import { SettingDto } from '@types';
import ROLES from '@constants/roles';

export interface IChangedParam extends Pick<SettingDto, 'clientAppCode' | 'key' | 'value' | 'userRole'> {
    type: SETTINGS_TYPES;
}

export const showNotify = (message: React.ReactNode, isError?: boolean) => {
    const config: ArgsProps = {
        duration: 5,
        message,
    };

    if (isError) {
        customNotifications.error(config);
    } else {
        customNotifications.success(config);
    }

};

type UpdateAndAddSettings = {
    updateSettings: SettingDto[];
    addSettingsArr: SettingDto[];
};

export async function createOrUpdateKey(changedParams: IChangedParam[]) {
    const { updateSettings, addSettingsArr } = changedParams.reduce<UpdateAndAddSettings>((result, { type, ...params }) => {
        const key = type === SETTINGS_TYPES.EDIT ? 'updateSettings' : 'addSettingsArr';

        if (params.key === 'referralTokenLifetime') {
            delete params.clientAppCode;
            params.key = 'token_lifetime';
            params.userRole = ROLES.REFERAL_LINK;
        }

        return {
            ...result,
            [key]: [...result[key], params],
        };
    }, { updateSettings: [], addSettingsArr: [] });

    updateSettings.length && (await updateSettingsList(updateSettings));
    addSettingsArr.length && (await addSettings(addSettingsArr));
}
