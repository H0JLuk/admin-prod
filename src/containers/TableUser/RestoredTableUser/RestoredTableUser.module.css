.titleRestored {
    font-weight: 700;
}

.table {
    width: 100%;
    min-width: 360px;
}

.usersList {
    margin-bottom: 10px;
    max-width: 350px;
}


.passwordColumn{
    text-align: right;
}

.error{
    color:red;
}

.buttonWrapper {
    text-align: center;
}

.button:global(.ant-btn){
    background: #09a552;
    border-color:#09a552;
    font-family: 'SBSansText';
    color: white;
}
