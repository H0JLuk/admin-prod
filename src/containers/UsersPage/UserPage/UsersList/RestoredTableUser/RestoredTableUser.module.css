.titleRestored {
    font-weight: 700;
}

.usersList {
    margin-bottom: 10px;
    max-width: 350px;
}

.buttonWrapper {
    text-align: center;
}
