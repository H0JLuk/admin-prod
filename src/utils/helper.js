import moment from 'moment';

export const downloadFile = (data, name) => {
    const blob = new Blob([data], { type: 'application/vnd.ms-excel' });
    downloadFileFunc(URL.createObjectURL(blob), name, 'xlsx');
};

export const loadImageWithPromise = (url, failUrl) => {
    return new Promise((resolve, reject) => {
        const img = new Image();
        img.onload = () => resolve(url);
        img.onerror = () => reject(failUrl);
        img.src = url;
    });
};

export function templateLink(filename) {
    return `${process.env.PUBLIC_URL}/templates/${filename}`;
}

export const getBase64 = (file) => {
    const reader = new FileReader();
    return new Promise((resolve, reject) => {
        try {
            reader.readAsDataURL(file);
            reader.onload = () => resolve(reader.result);
        } catch {
            reader.onerror = (error) => reject(error);
        }
    });
};

/**
 * @param {any[]} array
 * @param {number} from
 * @param {number} to
 */
export function arrayMove(array, from, to) {
    const newArray = [...array];
    const startIndex = from < 0 ? newArray.length + from : from;

    if (startIndex >= 0 && startIndex < newArray.length) {
        const endIndex = to < 0 ? newArray.length + to : to;
        const [item] = newArray.splice(from, 1);
        newArray.splice(endIndex, 0, item);
    }

    return newArray;
}

export function getFormattedDate(date, format = 'DD.MM.YYYY') {
    if (!date) {
        return '';
    }

    return moment(date).format(format);
}

/**
 * @param {string[][]} data
 * @param {string[]} columns
 */
export function generateCsvFile(data, columns = ['Табельный номер пользователя', 'Пароль']) {
    const contentType = 'data:text/csv;charset=utf-8,';
    const finalCsv = [columns, ...data].map(item => item.join(',')).join('\n');
    const encodedUri = encodeURI(contentType + finalCsv);
    downloadFileFunc(encodedUri, 'restoredUsers', 'csv');
}

export function downloadFileFunc(dataObj, name = 'file', fileExtension = 'csv') {
    if (dataObj) {
        const link = window.document.createElement('a');
        link.setAttribute('href', dataObj);
        link.setAttribute('download', `${name}.${fileExtension}`);
        document.body.appendChild(link);
        link.click();
        document.body.removeChild(link);
    }
}