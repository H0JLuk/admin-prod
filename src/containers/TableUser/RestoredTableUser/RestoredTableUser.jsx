import React from 'react';
import { Button } from 'antd';
import styles from './RestoredTableUser.module.css';
import { generateCsvFile } from '../../../utils/helper';

const TITLE = 'Пароль успешно сброшен для:';
const EXPORT = 'Экспорт паролей';

const RestoredTableUser = ({ users }) => {

    const exportCSV = () => {
        generateCsvFile(users.reduce((prev, item) => ([...prev, Object.values(item)]), []));
    };
    return (
        <>
            <p className={ styles.titleRestored }>
                { TITLE }
            </p>
            <p className={ styles.usersList }>
                { users.map(({ personalNumber }) => personalNumber).join(', ') }
            </p>
            <div className={ styles.buttonWrapper }>
                <Button onClick={ exportCSV } className={ styles.button }>
                    { EXPORT }
                </Button>
            </div>
        </>
    );
};


export default RestoredTableUser;