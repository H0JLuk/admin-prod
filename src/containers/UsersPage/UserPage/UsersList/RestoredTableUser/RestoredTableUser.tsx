import React from 'react';
import { UserInfo } from '@types';
import { Button } from 'antd';
import { generateCsvFile } from '../../../../../utils/helper';

import styles from './RestoredTableUser.module.css';

type RestoredUser = {
    personalNumber: UserInfo['personalNumber'];
    generatedPassword: string;
};

export type RestoredTableUserProps = {
    users: RestoredUser[];
};

const TITLE = 'Пароль успешно сброшен для:';
const EXPORT = 'Экспорт паролей';


const RestoredTableUser: React.FC<RestoredTableUserProps> = ({ users }) => {

    const exportCSV = () => {
        generateCsvFile(users.reduce((prev: string[][], item) => ([...prev, Object.values(item)]), []));
    };
    return (
        <>
            <p className={styles.titleRestored}>
                {TITLE}
            </p>
            <p className={styles.usersList}>
                {users.map(({ personalNumber }) => personalNumber).join(', ')}
            </p>
            <div className={styles.buttonWrapper}>
                <Button
                    type="primary"
                    onClick={exportCSV}
                >
                    {EXPORT}
                </Button>
            </div>
        </>
    );
};

export default RestoredTableUser;
