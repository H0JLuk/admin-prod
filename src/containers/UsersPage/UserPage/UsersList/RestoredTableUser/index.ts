import RestoredTableUser from './RestoredTableUser';

export type { RestoredTableUserProps } from './RestoredTableUser';

export default RestoredTableUser;
