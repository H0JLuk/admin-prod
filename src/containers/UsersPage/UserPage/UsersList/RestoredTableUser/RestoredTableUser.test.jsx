import { shallow } from 'enzyme';
import React from 'react';
import * as utils from '../../../../../utils/helper';
import RestoredTableUser from './RestoredTableUser';


describe('<RestoredTableUser /> tests', () => {
    const props = {
        users: [
            {
                generatedPassword: '1111111111',
                personalNumber: '333333333333',
            },
            {
                generatedPassword: '555555555555',
                personalNumber: '4444444444444',
            },
            {
                generatedPassword: '9999999999999',
                personalNumber: '2000000000000',
            },
        ],
    };

    it('RestoredTableUser shapshot', () => {
        const RestoredTableUserItem = shallow(<RestoredTableUser { ...props } />);
        expect(RestoredTableUserItem.html()).toMatchSnapshot();
    });

    it('should click export button', () => {
        utils.generateCsvFile = jest.fn();
        const RestoredTableUserItem = shallow(<RestoredTableUser { ...props } />);
        RestoredTableUserItem.find('Button').simulate('click');
        expect(utils.generateCsvFile).toBeCalled();
    });

    it('should render title', () => {
        const RestoredTableUserItem = shallow(<RestoredTableUser { ...props } />);
        expect(RestoredTableUserItem.find('.titleRestored').text()).toEqual('Пароль успешно сброшен для:');
    });

    it('should render usersList', () => {
        const RestoredTableUserItem = shallow(<RestoredTableUser { ...props } />);
        expect(RestoredTableUserItem.find('.usersList')).toHaveLength(1);
    });

});
