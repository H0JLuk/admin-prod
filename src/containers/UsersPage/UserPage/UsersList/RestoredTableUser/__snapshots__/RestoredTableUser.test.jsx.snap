// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RestoredTableUser /> tests RestoredTableUser shapshot 1`] = `"<p class=\\"titleRestored\\">Пароль успешно сброшен для:</p><p class=\\"usersList\\">333333333333, 4444444444444, 2000000000000</p><div class=\\"buttonWrapper\\"><button type=\\"button\\" class=\\"ant-btn ant-btn-primary\\"><span>Экспорт паролей</span></button></div>"`;
